Events from raw data file are added as annotations

kuziekj opened the pull request:
So events should now be read from the 'digaux' column (if present) in the raw data file.

This should also work for both types of data files (parsed and unparsed).

I want to add some tests for this before merging, which I'll work on.

I also figure at this point we should add tests for parsed and unparsed files, which I'll also add.

I'll also create a PR for the new test data once I get the data shortened and run through ppod.

kuziekj committed:
events from raw data file are added as annotations

kuziekj commented:
Also I noticed that when we send triggers using the 'digaux', a particular trigger is constantly recorded until a new trigger is sent or the old trigger is turned off.

For example, if we use the pi to send a trigger of '1', a '1' will be recorded for every sample point until a different trigger is sent, or until we turn all the pins on the pi off. Do we want to create annotations for every trigger (every '1'), or just to the first instance of a trigger (the first '1')?

kylemath commented:
> Also I noticed that when we send triggers using the 'digaux', a particular trigger is constantly recorded until a new trigger is sent or the old trigger is turned off.
> 
> For example, if we use the pi to send a trigger of '1', a '1' will be recorded for every sample point until a different trigger is sent, or until we turn all the pins on the pi off. Do we want to create annotations for every trigger (every '1'), or just to the first instance of a trigger (the first '1')?

This is more an issue with incorrect triggers, the correct way of sending triggers should be to briefly set the pins and then reset them to zero, this needs to be done with the brain products systems as well

kylemath reviewed (changes requested):
looks good to me, but not sure how to test,

what data files have you tested these with and do you have a test script somewhere that I can run to test it on my machine

kuziekj commented:
> looks good to me, but not sure how to test,
> what data files have you tested these with and do you have a test script somewhere that I can run to test it on my machine

I made some proper boxy recordings, with triggers sent to the 'digaux' yesterday, so I'll share those with you and make a PR for the new data.

I'm working on adding trigger tests to 'test_boxy.py', I should be able to push those changes later today.

> This is more an issue with incorrect triggers, the correct way of sending triggers should be to briefly set the pins and then reset them to zero, this needs to be done with the brain products systems as well

well sort of, the brain products only records a trigger when there's a change in the pins. The imagent machine works a bit different in that it's always recording the current state of the pins, not just when there's a change in the pins. You're right in that this could be avoided by quickly setting the pins back to zero before the next sample is recorded. I added a fix to this in 'boxy.py' so that it only considers the when the pins change. For example, if the 'digaux' looks like this (1, 1, 1, 1, 1, 2, 2, 2, 2, 2), it will ignore consecutive repeats and return annotations for two triggers (1, 2) rather than ten triggers (1, 1, 1, 1, 1, 2, 2, 2, 2, 2). I hope that makes sense.... At any rate, this will be added to the next PR for you to take a look.

kuziekj committed:
ignores conssecutive trigger repeats in digaux, added tests for filetypes and digaux

kuziekj commented:
Alright, I added tests for reading the digaux channel from both parsed and unparsed files, and comparing to how the digaux is read by p_pod. I also added a test to compare the loaded data between parsed and unparsed.

Consecutive repeats in digaux are also ignored, so only the first instance of a trigger is used but the repeats are used to determine the duration of the trigger.

I'll send you the test files I used. Once we are comfortable with this PR I'll make a PR for the test files.

kylemath commented:
this looks really great, have you loaded in the data and observed the events yourself to check?

kylemath commented:
is there like a dev/plot_40.py file that loads in some data and shows it? Not an official test file but a test file for us to try things out like we were using the tutorial before?

kuziekj commented:
Yeah I checked the events and everything seems fine. Used the pi to send triggers 1, 2, 3, 4, and 5, each lasting for 500ms and spaced 500ms apart. That seems to be what we're getting when reading the digaux.

There are a few tutorials that demonstrate loading in data and plotting it (tutorials/intro and tutorials/raw) but they seem more general. These tutorials seem concerned about using MNE and the raw data object in general. I'm surprised the tutorials regarding how to load specific types of data (tutorials/io) only talk about the data and don't have any actual code to load or plot the data.

Anyway, did you want me to make a tutorial for loading and plotting data, maybe just for our testing purposes?

kylemath commented:
Yes please

On Wed., Aug. 5, 2020, 9:46 a.m. kuziekj, <notifications@github.com> wrote:

> Yeah I checked the events and everything seems fine. Used the pi to send
> triggers 1, 2, 3, 4, and 5, each lasting for 500ms and spaced 500ms apart.
> That seems to be what we're getting when reading the digaux.
>
> There are a few tutorials that demonstrate loading in data and plotting it
> (tutorials/intro and tutorials/raw) but they seem more general. These
> tutorials seem concerned about using MNE and the raw data object in
> general. I'm surprised the tutorials regarding how to load specific types
> of data (tutorials/io) only talk about the data and don't have any actual
> code to load or plot the data.
>
> Anyway, did you want me to make a tutorial for loading and plotting data,
> maybe just for our testing purposes?
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub
> <https://github.com/kylemath/mne-python/pull/27#issuecomment-669270190>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AA36GFMRI2AFJ6IIPG3CBATR7F5GPANCNFSM4POKANDQ>
> .
>

kuziekj committed:
fixed some typos

kuziekj committed:
fixed more spelling and style errors

kuziekj commented:
@kylemath Do you think this is okay to merge now?

kylemath commented:
Yes lgtm
